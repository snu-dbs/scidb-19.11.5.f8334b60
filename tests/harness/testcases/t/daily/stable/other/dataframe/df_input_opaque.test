# Test OPAQUE input()/load() into dataframes.
--setup
--start-query-logging

store(apply(build(<v:int64>[i=0:199:0:10], i), (w, 2*v)), A)

# query logging suppressed for all queries that would expand ${HPID} into .expected
--stop-query-logging
save(A, '/tmp/df_input_${HPID}_array.opaque', format:'opaque')
save(project(flatten(A), v, w), '/tmp/df_input_${HPID}_frame.opaque', format:'opaque')
--start-query-logging

--test
--echo Baseline sums.
op_sum(A, v)
op_sum(flatten(A), v)
op_sum(A, w)
op_sum(flatten(A), w)

--echo Sums from input()
--stop-query-logging
--igdata "input(<v:int64, w:int64>, '/tmp/df_input_${HPID}_frame.opaque', format:'opaque')"
op_sum(input(<v:int64, w:int64>, '/tmp/df_input_${HPID}_frame.opaque', format:'opaque'), v)
op_sum(input(<v:int64, w:int64>, '/tmp/df_input_${HPID}_frame.opaque', format:'opaque'), w)
--start-query-logging

--echo Reload saved dataframe.
--stop-query-logging
--igdata "input(<v:int64, w:int64>, '/tmp/df_input_${HPID}_frame.opaque', format:'opaque')"
store(input(<v:int64, w:int64>, '/tmp/df_input_${HPID}_frame.opaque', format:'opaque'), DF)
--start-query-logging
# show(DF)
project(apply(show(DF), nschema, substr(schema, 0, 29)+'-'+substr(schema, 30, 100)), nschema)
op_sum(DF, v)
op_sum(DF, w)

--echo SG found reloading a dataframe saved in opaque format.
--stop-query-logging
--shell --store --command "iquery -c ${IQUERY_HOST} -p ${IQUERY_PORT} -aq \
    "_explain_physical( \
        'store(input(<v:int64, w:int64>, \'/tmp/df_input_${HPID}_frame.opaque\', format:\'opaque\'), DF)', \
        'afl')" | \
    awk '/impl_sg/ { print "Found SG"; found = 1 } END { if (!found) print "No SG found" }' "
--start-query-logging

--echo SG found storing an array.
--stop-query-logging
--shell --store --command "iquery -c ${IQUERY_HOST} -p ${IQUERY_PORT} -aq \
    "_explain_physical( \
        'store(input(A, \'/tmp/df_input_${HPID}_array.opaque\', format:\'opaque\'), B)', \
        'afl')" | \
    awk '/impl_sg/ { print "Found SG"; found = 1 } END { if (!found) print "No SG found" }' "
--start-query-logging

--cleanup
remove(A)
remove(DF)
--shell --command "rm -f /tmp/df_input_${HPID}_*.opaque"
