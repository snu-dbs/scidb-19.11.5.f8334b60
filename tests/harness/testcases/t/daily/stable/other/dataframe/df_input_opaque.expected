SCIDB QUERY : <store(apply(build(<v:int64>[i=0:199:0:10], i), (w, 2*v)), A)>
Query was executed successfully

{i} v,w

v,w

Baseline sums.
SCIDB QUERY : <op_sum(A, v)>
{i} v_sum
{0} 19900

SCIDB QUERY : <op_sum(flatten(A), v)>
{i} v_sum
{0} 19900

SCIDB QUERY : <op_sum(A, w)>
{i} w_sum
{0} 39800

SCIDB QUERY : <op_sum(flatten(A), w)>
{i} w_sum
{0} 39800

Sums from input()
[Query was executed successfully, ignoring data output by this query.]

{i} v_sum
{0} 19900

{i} w_sum
{0} 39800

Reload saved dataframe.
[Query was executed successfully, ignoring data output by this query.]

Query was executed successfully

SCIDB QUERY : <project(apply(show(DF), nschema, substr(schema, 0, 29)+'-'+substr(schema, 30, 100)), nschema)>
{i} nschema
{0} 'DF<v:int64,w:int64> [$inst=0:-:0:1; $seq=0:*:0:10]'

SCIDB QUERY : <op_sum(DF, v)>
{i} v_sum
{0} 19900

SCIDB QUERY : <op_sum(DF, w)>
{i} w_sum
{0} 39800

SG found reloading a dataframe saved in opaque format.
Found SG

SG found storing an array.
Found SG

SCIDB QUERY : <remove(A)>
Query was executed successfully

SCIDB QUERY : <remove(DF)>
Query was executed successfully

